server/utils/licenceStatusTransitions.js

```diff
@@ -1,170 +0,0 @@
-const { taskState } = require('../services/config/taskState')
-
-module.exports = { getAllowedTransition }
-
-function getAllowedTransition(licenceStatus, role) {
-  if (!licenceStatus) {
-    return null
-  }
-
-  switch (role) {
-    case 'RO':
-      if (canSendRoToCa(licenceStatus)) {
-        return 'roToCa'
-      }
-
-      return null
-
-    case 'CA':
-      if (canSendCaToDmRefusal(licenceStatus)) {
-        return 'caToDmRefusal'
-      }
-
-      if (canSendCaToDm(licenceStatus)) {
-        return 'caToDm'
-      }
-
-      if (canSendCaToRo(licenceStatus)) {
-        return 'caToRo'
-      }
-      return null
-
-    default:
-      return null
-  }
-}
-
-function canSendRoToCa(licenceStatus) {
-  const { tasks, stage, decisions } = licenceStatus
-
-  if (stage !== 'PROCESSING_RO') {
-    return false
-  }
-
-  if (decisions.bassReferralNeeded && tasks.bassAreaCheck === taskState.DONE) {
-    return true
-  }
-
-  if (decisions.curfewAddressRejected) {
-    return true
-  }
-
-  if (decisions.optedOut) {
-    return true
-  }
-
-  const required = [
-    tasks.curfewAddressReview,
-    tasks.curfewHours,
-    tasks.licenceConditions,
-    decisions.approvedPremisesRequired ? taskState.DONE : tasks.riskManagement,
-    tasks.victim,
-    tasks.reportingInstructions,
-  ]
-
-  return required.every((it) => it === taskState.DONE)
-}
-
-function canSendCaToRo(licenceStatus) {
-  const { tasks, decisions, stage } = licenceStatus
-
-  const { eligible, optedOut, bassReferralNeeded, curfewAddressRejected, approvedPremisesRequired } = decisions
-
-  if (['PROCESSING_CA', 'MODIFIED', 'MODIFIED_APPROVAL'].includes(stage)) {
-    if (bassReferralNeeded) {
-      if (licenceStatus.tasks.bassAreaCheck === taskState.UNSTARTED) {
-        return true
-      }
-    } else if (!optedOut && !approvedPremisesRequired && tasks.curfewAddressReview === taskState.UNSTARTED) {
-      return true
-    }
-  }
-
-  const notToProgress = !eligible || optedOut || curfewAddressRejected
-
-  if (stage !== 'ELIGIBILITY' || notToProgress) {
-    return false
-  }
-
-  const required = [tasks.exclusion, tasks.crdTime, tasks.suitability, tasks.optOut, tasks.curfewAddress]
-
-  const allTaskComplete = required.every((it) => it === taskState.DONE)
-
-  if (bassReferralNeeded) {
-    return allTaskComplete && tasks.bassReferral === taskState.DONE
```

Comment:
`task.bassReferral` isn't a task! 

I'm not quite sure what it was supposed to be but it's obviously never evaluating to true so removed from new version in `caTransitions`